is-state/is-state.typ

```diff
@@ -0,0 +1,227 @@
+    },
+    ...
+  ]
+}
+```
+Für jedes Modul wird anhand der `"zuordnungen[].url"` eine Anfrage gemacht, woraus das folgende Feld relevant ist.
+```json
+{
+  "kreditpunkte": 4
+}
+```
+Für jede Spezialisierung wird anhand der `"spezialisierungen[].url"` eine Anfrage gemacht, woraus die folgenden Felder relevant sind.
+```json
+{
+  "zuordnungen": [
+    {
+      "kuerzel": "M_AppArch"
+    },
+    ...
+  ]
+}
+```
+
+Das Diagram @crawler stellt den Ablauf, bei dem der Crawler die oben erwähnten Daten verarbeitet, grob dar.
+#figure(image("Crawler.drawio.png"), caption: [Ablauf des Crawlers]) <crawler>
+
+Wie bereits erwähnt, wird das Format der Daten für die weitere Nutzung angepasst.
+
+Module werden in folgendem Format gespeichert.
+```json
+[
+  {
+    "id": "AppArch",
+    "name": "Application Architecture",
+    "url": "allModules/28236_M_AppArch.json",
+    "isThesis": false,
+    "isRequired": false,
+    "recommendedSemester": 7,
+    "ects": 4,
+    "categories_for_coloring": [
+      "Auf",
+      "Inf"
+    ]
+  },
+  ...
+]
+```
+
+Kategorien in folgendem Format.
+```json
+[
+  {
+    "id": "Auf",
+    "required_ects": 48,
+    "name": "Aufbau",
+    "modules": [
+      {
+        "id": "AppArch",
+        "name": "Application Architecture",
+        "url": "allModules/28236_M_AppArch.json"
+      },
+      ...
+    ]
+  },
+  ...
+]
+```
+
+Spezialisierungen in folgendem Format.
+```json
+[
+  {
+    "id": "FrEng",
+    "url": "allStudies/10193_FrEng.json",
+    "name": "Frontend Engineering",
+    "modules": [
+      {
+        "id": "AppArch",
+        "name": "Application Architecture",
+        "url": "allModules/28236_M_AppArch.json"
+      },
+      ...
+    ]
+  },
+  ...
+]
+```
+
+=== Code
+
+Die Applikation selbst ist eine #gls("spa"), geschrieben in Vue.
+Es wird Typescript anstatt JavaScript verwendet.
+Die Icons kommen von Fontawesome, das Styling wird grösstenteils über Tailwind gemacht.
+Als Build Tool wird Vite eingesetzt.
+
+Vor dem Start unserer Arbeit wurde mit dem Stakeholder, welcher auch Haupt-Maintainer ist, besprochen, dass sich dieser TechStack nicht durch unsere Arbeit verändern soll.
+
+=== Hosting
+
+// Glossar or Bib? https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#about-github-pages
+Die Applikation wird über GitHub Pages gehosted.
+Da sie lediglich eine #gls("spa") ohne Backend ist, entfallen somit jegliche Kosten für das Hosting.
+
+// I want the steps to be styled differently
+Die gewünschte URL, #link("lost.university"), wird auf GitHub unter Settings -> Pages -> Custom Domain hinterlegt.
+Bei der Domain selbst wird die Repository als ein CNAME hinterlegt.
+
+// I want "main" to be styled differently
+// Glossar or Bib? https://docs.github.com/en/actions/about-github-actions
+Wird ein Branch in den main Branch gemergt, wird über einen GitHub Workflow eine GitHub Action ausgelöst, welche die #gls("spa") buildet und deployed.
```

Comment:
```suggestion
Wird ein Branch in den main-Branch gemergt, wird über einen GitHub Workflow eine GitHub Action ausgelöst, welche die #gls("spa") baut und deployt.
```

Reply:
anglizisme sind en PAIN. uf jede fall isches im dütsche immer es "t" am schluss und nie es "ed" wie im englische. bi immer chli am google wie die einzelne wörter vo andere website brucht werded. wenns gaht, würdi dütschi wörter bruche aber für deploye chunnt mer keis gschids in sinn (undem deepl/pons wörterbuech au nöd)

Reply:
Evtl eher _"main" Branch_ anstatt _main-Branch_?

Reply:
GPT isch au für "main-Branch".